solidity/contracts/converter/types/standard-pool/StandardPoolConverter.sol

```diff
@@ -0,0 +1,1362 @@
+        uint256 amount = targetAmount - fee;
+
+        // ensure that the trade gives something in return
+        require(amount != 0, "ERR_ZERO_TARGET_AMOUNT");
+
+        // ensure that the trade won't deplete the reserve balance
+        assert(amount < targetBalance);
+
+        // ensure that the input amount was already deposited
+        uint256 actualSourceBalance;
+        if (_sourceToken == ETH_RESERVE_ADDRESS) {
+            actualSourceBalance = address(this).balance;
+            require(msg.value == _amount, "ERR_ETH_AMOUNT_MISMATCH");
+        }
+        else {
+            actualSourceBalance = _sourceToken.balanceOf(address(this));
+            require(msg.value == 0 && actualSourceBalance.sub(sourceBalance) >= _amount, "ERR_INVALID_AMOUNT");
+        }
+
+        // sync the reserve balances
+        writeReserveBalances(sourceId, targetId, actualSourceBalance, targetBalance - amount);
+
+        // transfer funds to the beneficiary in the to reserve token
+        if (_targetToken == ETH_RESERVE_ADDRESS)
+            _beneficiary.transfer(amount);
+        else
+            safeTransfer(_targetToken, _beneficiary, amount);
+
+        // dispatch the conversion event
+        dispatchConversionEvent(_sourceToken, _targetToken, _trader, _amount, amount, fee);
+
+        // dispatch rate updates
+        dispatchTokenRateUpdateEvents(_sourceToken, _targetToken, actualSourceBalance, targetBalance - amount);
+
+        return amount;
+    }
+
+    /**
+      * @dev returns the recent average rate of 1 `_token` in the other reserve token units
+      *
+      * @param _token   token to get the rate for
+      * @return recent average rate between the reserves (numerator)
+      * @return recent average rate between the reserves (denominator)
+    */
+    function recentAverageRate(IERC20Token _token) external view returns (uint256, uint256) {
+        // get the recent average rate of reserve 0
+        uint256 rate = calcRecentAverageRate(averageRateInfo);
+        if (_token == __reserveTokens[0]) {
+            return ((rate >> 112) & MAX_UINT112, rate & MAX_UINT112);
+        }
+
+        return (rate & MAX_UINT112, (rate >> 112) & MAX_UINT112);
+    }
+
+    /**
+      * @dev updates the recent average rate if needed
+    */
+    function updateRecentAverageRate() internal {
+        uint256 averageRateInfo1 = averageRateInfo;
+        uint256 averageRateInfo2 = calcRecentAverageRate(averageRateInfo1);
+        if (averageRateInfo1 != averageRateInfo2) {
+            averageRateInfo = averageRateInfo2;
+        }
+    }
+
+    /**
+      * @dev returns the recent average rate of 1 reserve token 0 in reserve token 1 units
+      *
+      * @param _prevAverageRate a local copy of the previous average rate state-variable
+      * @return recent average rate between the reserves
+    */
+    function calcRecentAverageRate(uint256 _prevAverageRate) internal view returns (uint256) {
```

Comment:
`_prevAverageRate` is the average rate info, not only previous rate

Reply:
Fixed.